File: sorting/insertionsort.js
let arr = [123, 42, 12, 9, 400, 23, 1, 0, 21];

function insertionSort(arr) {
for (i = 0; i < arr.length; i++) {
        let temp = arr[i];
        let j = i -1;
        while (arr[j] > temp && j >= 0) {
            arr[j+1] = arr[j];
            j--;
        }
        arr[j+1] = temp
    }
    return arr;
}

console.log(insertionSort(arr));

File: sorting/foundational/insertion_sort.py
arr = [123, 42, 12, 9, 400, 23, 1, 0, 21]

def insertionSort(arr):
    for i in range(0, len(arr)):
            temp = arr[i]
            j = i -1
            while (arr[j] > temp and j >= 0):
                arr[j+1] = arr[j]
                j-=1
            arr[j+1] = temp

    return arr


print(insertionSort(arr))